text/0000-named-tracers.md

```diff
@@ -0,0 +1,91 @@
+* [Global Instance discussions](https://github.com/open-telemetry/opentelemetry-specification/labels/global%20instance)
+* [Proposal: Add a version resource](https://github.com/open-telemetry/oteps/pull/38)
+
+## Motivation
+
+The mechanism of "Named Tracers" proposed here is motivated by following scenarios:
+
+* For a consumer of OpenTelemetry instrumentation libraries, there is currently no possibility of influencing the amount of the data produced by such libraries. Instrumentation libraries can easily "spam" backend systems, deliver bogus data or - in the worst case - crash or slow down applications. These problems might even occur suddenly in production environments caused by external factors such as increasing load or unexpected input data.
+
+* If a library hasn't implemented [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md) correctly or those conventions change over time, it's currently hard to interpret and sanitize these data selectively. The produced Spans cannot associated with those instrumentation libraries later.
+
+This proposal attempts to solve the stated problems by introducing the concept of:
+ * _Named Tracers_ identified via a Resource (e.g. _"io.opentelemetry.contrib.mongodb"_, _"1.0.0"_) which is associated with the Tracer and the Spans it produces.
+ * A `TracerFactory` as the only means of creating a Tracer.
+
+Based on such an identifier, a Sampler could be implemented that discards Spans from certain libraries. Also, by providing a custom Exporter, Span data could be sanitized before it gets processed in a back-end system. However, this is beyond the scope of this proposal, which only provides the fundamental mechanisms.
+
+## Explanation
+
+From a user perspective, working with *Named Tracers* and *TracerFactories* is conceptually similar to how e.g. the [Java logging API](https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#getLogger(java.lang.String)) and logging frameworks like [log4j](https://www.slf4j.org/apidocs/org/slf4j/LoggerFactory.html) work. In analogy to requesting Logger objects through LoggerFactories, a tracing library would create specific 'Tracer' objects through a 'TracerFactory'.
+
+Instead of using plain strings as an argument for creating new Tracers a Resource identifying an instrumentation library is used. Such resources must have a _version_ and _name_ labels (there could be semantic convention definitions for those labels). Version values will follow the conventions proposed [here](https://github.com/open-telemetry/oteps/pull/38).
+
+
+```java
+// Create resource for given instrumentation library information (name + version)
+Map<String, String> libraryLabels = new HashMap<>();
+libraryLabels.put("name", "io.opentelemetry.contrib.mongodb");
+libraryLabels.put("version", "1.0.0");
```

Comment:
Why `version` for component? If you have a version for the system as an application owner you can determine all the components versions (I hope everyone is using reproducible builds).

Based on the rationals that I see in this OTEP, I do not understand the need for a `version` (but I do clearly understand the need for a `name`).

Reply:
Some examples:
* the details of how to collect less data from certain named traces is not really described here, but it's easy to imagine a scenario where some specific range of versions of a given middleware needs to be suppressed, while other versions allowed (perhaps because of a bug in the instrumentation). Suppression may need to happen online, which makes resolving version from the build number again very complicated.
* go modules allows multiple versions of the same middleware in a single build, so you cannot resolve version just from the build

Reply:
I also see the usefulness of having this version information for the reasons above. We could however consider making this part of the "name" (Resource, ...) explicitly optional?

Reply:
Please document the reasons if you want to keep the version.

@yurishkuro based on my understanding you can have multiple versions only if they are different major versions which implies different package names (which probably implies different names).

@yurishkuro @z1c0 I see the advantage from the first bullet point that @yurishkuro posted. So I am happy to keep the version, but make sure it is documented in the requirements so we understand why we need version.